README.md

```diff
@@ -1,54 +1,142 @@
-# Awesome Advice [![awesome.list](https://cdn.rawgit.com/sindresorhus/awesome/d7305f38d29fed78fa85652e3a63e154dd8e8829/media/badge.svg)](https://github.com/sindresorhus/awesome) 
+# Awesome Advice [![awesome.list](https://cdn.rawgit.com/sindresorhus/awesome/d7305f38d29fed78fa85652e3a63e154dd8e8829/media/badge.svg)](https://github.com/sindresorhus/awesome)
 
 > A collection of awesome pieces of advice
 
-### priority
-- the better the problem is defined, the better the solution will be
+## Priority
+- The better the problem is defined, the better the solution will be
 
-#### research
+## Research
 - #1. most important thing in programming is knowing how to research.
 - #2. is research.
 - #3. is research.
 
-#### 15 minute rule
+## 15 minute rule
   > general rule of thumb / proverbial saying, not 100% black-and-white, just don't be an askhole.
 
-  - if you ask for help on a problem before doing at least 15 minutes of work researching, debugging, and defining your problem, you're doing the other person a disservice.
-  - if you wait longer than 45 minutes and you are stuck, you are doing yourself a disservice.
+  - If you ask for help on a problem before doing at least 15 minutes of work researching, debugging, and defining your problem, you're doing the other person a disservice.
+  - If you wait longer than 45 minutes and you are stuck, you are doing yourself a disservice.
 
-#### proverbial
-- measure twice, cut once
-- there is never any code that is more permanent than temporary code
-- when you have a hammer, everything looks like a nail
-- if you're not using a framework, you're making one.
+## Proverbial
+- Measure twice, cut once
+- There is never any code that is more permanent than temporary code
+- When you have a hammer, everything looks like a nail
+- If you're not using a framework, you're making one.
 
-#### general
-- design patterns - don't let the names dictate the form of your code
-- expressive, clearly communicative, clean code
-  - have [variable names describe their intention](https://twitter.com/svensauleau/status/856424137493008384)
-- join in the community, conversing with people interested in the same thing, helping solve problems & getting your problems solved
-- don't worry if you think it sucks, contributing open source code helps build important skills, practice makes permanent
+## Principles (search these out if you don't know them!)
```

Comment:
Love it - thank you for putting this content in here! 1 minor change: Would you mind putting these Principles into their own `.md` file? `Principles.md`, then link to that?

Reply:
Would `Laws` also go in there, or separate file, or leave them here?

Reply:
Laws can go there too. 
A better name may be `best-practices.md`.